### ISSUE TITLE
Build failure with GHC 7.10

### ISSUE THREAD

**Issue by snoyberg** [juhp/cabal-rpm#18]
Some of the output is:

```
Unpacking to cabal-rpm-0.9.5/
Resolving dependencies...
Configuring cabal-rpm-0.9.5...
Building cabal-rpm-0.9.5...
Preprocessing executable 'cblrpm' for cabal-rpm-0.9.5...
[ 1 of 13] Compiling SysCmd           ( src/SysCmd.hs, dist/build/cblrpm/cblrpm-tmp/SysCmd.o )

src/SysCmd.hs:34:1: Warning:
    The import of ‘Data.Functor’ is redundant
      except perhaps to import instances from ‘Data.Functor’
    To import instances alone, use: import Data.Functor()
[ 2 of 13] Compiling FileUtils        ( src/FileUtils.hs, dist/build/cblrpm/cblrpm-tmp/FileUtils.o )

src/FileUtils.hs:27:1: Warning:
    The import of ‘Control.Applicative’ is redundant
      except perhaps to import instances from ‘Control.Applicative’
    To import instances alone, use: import Control.Applicative()
[ 3 of 13] Compiling Paths_cabal_rpm  ( dist/build/autogen/Paths_cabal_rpm.hs, dist/build/cblrpm/cblrpm-tmp/Paths_cabal_rpm.o )
[ 4 of 13] Compiling Setup            ( src/Setup.hs, dist/build/cblrpm/cblrpm-tmp/Setup.o )
[ 5 of 13] Compiling PackageUtils     ( src/PackageUtils.hs, dist/build/cblrpm/cblrpm-tmp/PackageUtils.o )

src/PackageUtils.hs:40:1: Warning:
    The import of ‘Control.Applicative’ is redundant
      except perhaps to import instances from ‘Control.Applicative’
    To import instances alone, use: import Control.Applicative()
[ 6 of 13] Compiling Dependencies     ( src/Dependencies.hs, dist/build/cblrpm/cblrpm-tmp/Dependencies.o )

src/Dependencies.hs:28:1: Warning:
    The import of ‘Control.Applicative’ is redundant
      except perhaps to import instances from ‘Control.Applicative’
    To import instances alone, use: import Control.Applicative()
[ 7 of 13] Compiling Commands.Spec    ( src/Commands/Spec.hs, dist/build/cblrpm/cblrpm-tmp/Commands/Spec.o )
```

**Comment by juhp** [comment]
Thanks for the report.

The full buildlog would be really helpful.

**Comment by snoyberg** [comment]
```
Unpacking to cabal-rpm-0.9.5/
Resolving dependencies...
Configuring cabal-rpm-0.9.5...
Building cabal-rpm-0.9.5...
Preprocessing executable 'cblrpm' for cabal-rpm-0.9.5...
[ 1 of 13] Compiling SysCmd           ( src/SysCmd.hs, dist/build/cblrpm/cblrpm-tmp/SysCmd.o )

src/SysCmd.hs:34:1: Warning:
    The import of ‘Data.Functor’ is redundant
      except perhaps to import instances from ‘Data.Functor’
    To import instances alone, use: import Data.Functor()
[ 2 of 13] Compiling FileUtils        ( src/FileUtils.hs, dist/build/cblrpm/cblrpm-tmp/FileUtils.o )

src/FileUtils.hs:27:1: Warning:
    The import of ‘Control.Applicative’ is redundant
      except perhaps to import instances from ‘Control.Applicative’
    To import instances alone, use: import Control.Applicative()
[ 3 of 13] Compiling Paths_cabal_rpm  ( dist/build/autogen/Paths_cabal_rpm.hs, dist/build/cblrpm/cblrpm-tmp/Paths_cabal_rpm.o )
[ 4 of 13] Compiling Setup            ( src/Setup.hs, dist/build/cblrpm/cblrpm-tmp/Setup.o )
[ 5 of 13] Compiling PackageUtils     ( src/PackageUtils.hs, dist/build/cblrpm/cblrpm-tmp/PackageUtils.o )

src/PackageUtils.hs:40:1: Warning:
    The import of ‘Control.Applicative’ is redundant
      except perhaps to import instances from ‘Control.Applicative’
    To import instances alone, use: import Control.Applicative()
[ 6 of 13] Compiling Dependencies     ( src/Dependencies.hs, dist/build/cblrpm/cblrpm-tmp/Dependencies.o )

src/Dependencies.hs:28:1: Warning:
    The import of ‘Control.Applicative’ is redundant
      except perhaps to import instances from ‘Control.Applicative’
    To import instances alone, use: import Control.Applicative()
[ 7 of 13] Compiling Commands.Spec    ( src/Commands/Spec.hs, dist/build/cblrpm/cblrpm-tmp/Commands/Spec.o )

src/Commands/Spec.hs:61:28:
    Couldn't match expected type ‘time-1.5.0.1:Data.Time.Format.Locale.TimeLocale’
                with actual type ‘System.Locale.TimeLocale’
    NB: ‘time-1.5.0.1:Data.Time.Format.Locale.TimeLocale’
          is defined in ‘Data.Time.Format.Locale’ in package ‘time-1.5.0.1’
        ‘System.Locale.TimeLocale’
          is defined in ‘System.Locale’ in package ‘old-locale-1.0.0.7’
    In the first argument of ‘formatTime’, namely ‘defaultTimeLocale’
    In the expression: formatTime defaultTimeLocale "0.%Y%m%d" now

src/Commands/Spec.hs:117:27:
    Couldn't match expected type ‘time-1.5.0.1:Data.Time.Format.Locale.TimeLocale’
                with actual type ‘System.Locale.TimeLocale’
    NB: ‘time-1.5.0.1:Data.Time.Format.Locale.TimeLocale’
          is defined in ‘Data.Time.Format.Locale’ in package ‘time-1.5.0.1’
        ‘System.Locale.TimeLocale’
          is defined in ‘System.Locale’ in package ‘old-locale-1.0.0.7’
    In the first argument of ‘formatTime’, namely ‘defaultTimeLocale’
    In the expression: formatTime defaultTimeLocale "%Y" now

src/Commands/Spec.hs:337:27:
    Couldn't match expected type ‘time-1.5.0.1:Data.Time.Format.Locale.TimeLocale’
                with actual type ‘System.Locale.TimeLocale’
    NB: ‘time-1.5.0.1:Data.Time.Format.Locale.TimeLocale’
          is defined in ‘Data.Time.Format.Locale’ in package ‘time-1.5.0.1’
        ‘System.Locale.TimeLocale’
          is defined in ‘System.Locale’ in package ‘old-locale-1.0.0.7’
    In the first argument of ‘formatTime’, namely ‘defaultTimeLocale’
    In the expression: formatTime defaultTimeLocale "%a %b %e %Y" now
[12 of 13] Compiling Commands.Depends ( src/Commands/Depends.hs, dist/build/cblrpm/cblrpm-tmp/Commands/Depends.o )

src/Commands/Depends.hs:25:1: Warning:
    The import of ‘Control.Applicative’ is redundant
      except perhaps to import instances from ‘Control.Applicative’
    To import instances alone, use: import Control.Applicative()

src/Commands/Depends.hs:28:24: Warning: Tab character
```

**Comment by juhp** [comment]
Thanks again

This should be fixed in cabal-rpm-0.9.5.1 .

I also added a travis multi-ghc configuration now,
which should help test the ghc portability from now on.
